app/lib/getPharmacies.js

```diff
@@ -41,7 +47,41 @@ function nearby(searchPoint, geo, limit) {
   const sortedOrgs = nearbyOrgs.sort(sortByDistance);
   console.timeEnd('sort-nearby-orgs');
 
-  return sortedOrgs.slice(0, maxResults);
+  const numberOfOpenToReturn = 3;
+  let numberReturned = 0;
+
+  console.time('filter-open-orgs');
+  const openServices = sortedOrgs.filter((item) => {
+    // TODO: Look at breaking out of the loop once three have been found
+    const openingTimes = item.openingTimes;
+    let isOpen;
+    let openingTimesMessage;
+
+    if (openingTimes) {
+      const openingTimesMoment = new OpeningTimes(item.openingTimes.general, 'Europe/London');
+      openingTimesMessage = openingTimesMoment.getOpeningHoursMessage(moment());
+      isOpen = openingTimesMoment.isOpen(moment());
+    } else {
+      openingTimesMessage = 'Call for opening times';
+      isOpen = false;
+    }
+    /* eslint-disable no-param-reassign */
+    item.openingTimesMessage = openingTimesMessage;
```

Comment:
There seems to be some duplication going on. In the services middleware getDisplayValuesMapper is handling the population of the opening times message. There probably is no need for getDisplayValuesMapper now but some of the functionality needs moving to here.